### PR TITLE
Forget legacy data dir

### LINKED ISSUE #4659: data_dir always falling back to legacy_data_dir
## Description

I'm not sure if I'm misunderstanding how I'm supposed to access the new pooch-handled data... but it seems that if all goes according to plan with the download in `data.__init__`, the `data_dir` variable will always fall back on the `legacy_data_dir`.  (i.e. if there's no break statement in the following clause, it will go to the else statement):

https://github.com/scikit-image/scikit-image/blob/f5290e1d0f10426dc8c54dc3046f795b4c2016bc/skimage/data/__init__.py#L168-L175

am I missing something?

PS.  thank you and congrats on the new release!

## Version information
3.8.2 | packaged by conda-forge | (default, Mar 23 2020, 17:55:48) 
[Clang 9.0.1 ]
macOS-10.13.6-x86_64-i386-64bit
scikit-image version: 0.17.0
numpy version: 1.18.4

### PR DESCRIPTION
**Pull request by jni**
## Description

Fixes #4659 

... I think. =P

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by tlambert03** [comment]
thanks!

**Comment by hmaarrfk** [comment]
Nice touch up

**Comment by hmaarrfk** [comment]
I think the problem is that some logic around making pooch optional got lost.

I opened an issue for it.

**Comment by hmaarrfk** [comment]
@meeseeksdev backport to v0.17.x

File: skimage/data/tests/test_data.py [hunk 1]
```diff
@@ -1,7 +1,16 @@
+import os
+from urllib import request
+
 import numpy as np
 import skimage.data as data
 from skimage import io
-from skimage._shared.testing import assert_equal, assert_almost_equal, fetch
+from skimage._shared.testing import (
+    assert_equal, assert_almost_equal, fetch, skipif
```
Comment (hmaarrfk) [inline skimage/data/tests/test_data.py:8]: using fetch from here will automatically skip things if it is offline.

Comment (hmaarrfk) [reply skimage/data/tests/test_data.py:8]: or i guess, if pooch isn't installed. We can add an offline check too.

**Pull request merged**